### PR TITLE
Update tree query to take execution space argument

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by dalg24** [comment]
I dropped commits that were getting rid of the `DeviceType` template argument for `Details:: BatchedQueries` because we run into the same issue where we need overloads that cast the views to const the data type (see https://github.com/arborx/ArborX/pull/244#issuecomment-601882373)

We can revisit later.

File: src/details/ArborX_DetailsSortUtils.hpp [hunk 1]
```diff
@@ -90,6 +88,8 @@ sortObjects(ViewType &view)
       bin_sort(view, CompType(n / 2, result.min_val, result.max_val), true);
   bin_sort.create_permute_vector();
   bin_sort.sort(view);
+  // FIXME Kokkos::BinSort is currently missing overloads that an execution
+  // space as argument
```
Comment (dalg24) [inline src/details/ArborX_DetailsSortUtils.hpp]: Need to report that on the Kokkos side and fix

File: src/details/ArborX_DetailsSortUtils.hpp [hunk 2]
```diff
@@ -98,18 +98,21 @@ sortObjects(ViewType &view)
...
-  ArborX::iota(Kokkos::Cuda{}, permute);
+  ArborX::iota(space, permute);
+
+  auto const execution_policy = thrust::cuda::par.on(space.cuda_stream());
 
   auto permute_ptr = thrust::device_ptr<size_t>(permute.data());
   auto begin_ptr = thrust::device_ptr<ValueType>(view.data());
   auto end_ptr = thrust::device_ptr<ValueType>(view.data() + n);
-  thrust::sort_by_key(begin_ptr, end_ptr, permute_ptr);
+  thrust::sort_by_key(execution_policy, begin_ptr, end_ptr, permute_ptr);
```
Comment (dalg24) [inline src/details/ArborX_DetailsSortUtils.hpp]: Making sure you see saw that

File: src/details/ArborX_DetailsSortUtils.hpp (hunk 1 above)
Comment (aprokop) [reply src/details/ArborX_DetailsSortUtils.hpp]: Please link it here when you open the issue there.

**Pull request merged**